File: Programming Basics with Python/По-сложни проверки/ski_trip.py

days_staying = int(input())
type_accommodation = input()
grade = input()

overnight_stays = days_staying - 1

if type_accommodation == "room for one person":
    overnight_stays_price = overnight_stays * 18
elif type_accommodation == "apartment":
    overnight_stays_price = overnight_stays * 25
    if days_staying < 10:
        overnight_stays_price *= 0.70
    elif days_staying < 15:
        overnight_stays_price *= 0.65
    else:
        overnight_stays_price *= 0.50
elif type_accommodation == "president apartment":
    overnight_stays_price = overnight_stays * 35
    if days_staying < 10:
        overnight_stays_price *= 0.90
    elif days_staying < 15:
        overnight_stays_price *= 0.85
    else:
        overnight_stays_price *= 0.80

if grade == "positive":
   overnight_stays_price *= 1.25
elif grade == "negative":
    overnight_stays_price *= 0.90

print(f"{overnight_stays_price:.2f}")

File: Programming Basics with Java/More complex checks/SkiTrip.java
import java.util.Scanner;

public class SkiTrip 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        int days = Integer.parseInt(input.nextLine());
        String premises = input.nextLine();
        String grade = input.nextLine();

        days -= 1;

        double percent = 0.0;
        double premises_price = 0.0;

        if (premises.equals("room for one person"))
        {
            premises_price = 18;
        }
        else if (premises.equals("apartment"))
        {
            premises_price = 25;
        }
        else if (premises.equals("president apartment"))
        {
            premises_price = 35;
        }

        if (days < 10)
        {
            if (premises.equals("apartment"))
            {
                percent = 0.7;
            }
            else if (premises.equals("president apartment"))
            {
                percent = 0.9;
            }
        }
        else if (days <= 15)
        {
            if (premises.equals("apartment"))
            {
                percent = 0.65;
            }
            else if (premises.equals("president apartment"))
            {
                percent = 0.85;
            }
        }
        else
        {
            if (premises.equals("apartment"))
            {
                percent = 0.5;
            }
            else if (premises.equals("president apartment"))
            {
                percent = 0.8;
            }
        }

        double price = 0.0;

        if (percent != 0.0)
        {
            price = (days * premises_price) * percent;
        }
        else
        {
            price = days * premises_price;
        }

        if (grade.equals("positive"))
        {
            price *= 1.25;
        }
        else
        {
            price *= 0.9;
        }

        System.out.printf("%.2f", price);
    }
}
